Pin Sphinx to 2.4.4 (take 2), fix docs CIs

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36072 Pin Sphinx to 2.4.4 (take 2), fix docs CIs**

Update to https://github.com/pytorch/pytorch/pull/36065/ which was
almost there

Test Plan:
- Wait for CI

Differential Revision: [D20871661](https://our.internmc.facebook.com/intern/diff/D20871661)

zou3519 committed:
Pin Sphinx to 2.4.4 (take 2), fix docs CIs

Update to https://github.com/pytorch/pytorch/pull/36065/ which was
almost there

Test Plan:
- Wait for CI

[ghstack-poisoned]

yf225 commented:
If `breathe` is still complaining, I suspect that we might need to pin the version of `breathe` as well :/

zou3519 committed:
Update on "Pin Sphinx to 2.4.4 (take 2), fix docs CIs"


Update to https://github.com/pytorch/pytorch/pull/36065/ which was
almost there

Test Plan:
- Wait for CI

Differential Revision: [D20871661](https://our.internmc.facebook.com/intern/diff/D20871661)

[ghstack-poisoned]

zou3519 commented:
@yf225 it looks like we don't need to pin breathe yet. However they're probably going to release a new release that is compatible with sphinx 3. So as a precaution we should definitely also pin it in a separate PR
